base/Base/Base_Base_implementation.F90

```diff
@@ -3307,17 +3307,16 @@ function grid_is_ok(grid) result(OK)
        call MAPL_GridGetInterior(grid,I1,I2,J1,J2)
        OK = .true. 
        ! check the edge of face 1 along longitude
+       allocate(corner_lons(I2-I1+2, J2-J1+2))
+       allocate(corner_lats(I2-I1+2, J2-J1+2))
+       call MAPL_GridGetCorners(Grid,corner_lons,corner_lats)
```

Comment:
Having the call MAPL_GridGetCorners outside the if statement solves the run time error when calling  the subroutine MAPL_GetGlobalHorzIJIndex.   Would we still enforce the deallocate statement here?

Reply:
I asked @tclune about this and I think since Fortran automatically deallocates when you leave a procedure, it's not necessary.

Reply:
The only time an explicit deallocate is required for a _local_ ALLOCATABLE variable is when an explicit allocation happens multiple times.  E.g., inside a loop.

Sometimes you may do an explicit deallocate to restore memory early if you are doing something where large memory use is a concern.

Usually, you just let the compiler do it when the variable goes out of scope at the end of the procedure.